ibc-clients/ics07-tendermint/src/client_state/common.rs

```diff
@@ -301,17 +299,18 @@ pub fn verify_membership<H: HostFunctionsProvider>(
     value: Vec<u8>,
 ) -> Result<(), ClientError> {
     if prefix.is_empty() {
-        return Err(ClientError::Ics23Verification(
+        return Err(ClientError::FailedIcs23Verification(
             CommitmentError::EmptyCommitmentPrefix,
         ));
     }
 
     let merkle_path = MerklePath::new(vec![prefix.as_bytes().to_vec().into(), path]);
-    let merkle_proof = MerkleProof::try_from(proof).map_err(ClientError::InvalidCommitmentProof)?;
+    let merkle_proof =
+        MerkleProof::try_from(proof).map_err(ClientError::FailedIcs23Verification)?;
```

Comment:
We can add the following `From` implementation under the `ClientError`, allowing us to move away from `map_err`s here.

```rust
impl From<CommitmentError> for ClientError {
    fn from(e: CommitmentError) -> Self {
        Self::FailedIcs23Verification(e)
    }
}
```